include/gpioMonitor.hpp

```diff
@@ -89,4 +92,44 @@ class GpioEventHandler
     const std::string& m_inventoryPath;
 };
 
+class GpioMonitor
+{
+  public:
+    GpioMonitor() = delete;
+    ~GpioMonitor() = default;
+    GpioMonitor(const GpioMonitor&) = delete;
+    GpioMonitor& operator=(const GpioMonitor&) = delete;
+    GpioMonitor(GpioMonitor&&) = delete;
+    GpioMonitor& operator=(GpioMonitor&&) = delete;
+
+    /**
+     * @brief constructor
+     *
+     * @param[in] i_sysCfgJsonObj - System config JSON Object.
+     * @param[in] i_ioContext - pointer to io context object.
+     */
+    GpioMonitor(const nlohmann::json& i_sysCfgJsonObj,
+                const std::shared_ptr<boost::asio::io_context>& i_ioContext) :
+        m_sysCfgJsonObj(i_sysCfgJsonObj)
+    {
+        initGpioInfos(i_ioContext);
+    }
+
+  private:
+    /**
+     * @brief API to extract GPIO information from JSON.
```

Comment:
Brief is not aligned with what is intended to be done in the function.

Reply:
Updated